src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs

```diff
@@ -2583,18 +2583,30 @@ internal override NamedTypeSymbol TryLookupForwardedMetadataTypeWithCycleDetecti
                 {
                     var peModuleSymbol = (Metadata.PE.PEModuleSymbol)_modules[i];
 
-                    var forwardedToAssembly = peModuleSymbol.GetAssemblyForForwardedType(ref emittedName);
-                    if ((object)forwardedToAssembly != null)
+                    (AssemblySymbol firstSymbol, AssemblySymbol secondSymbol) = peModuleSymbol.GetAssembliesForForwardedType(ref emittedName);
+
+                    if ((object)firstSymbol != null)
                     {
+                        if ((object)secondSymbol != null)
+                        {
+                            var forwardingErrorInfo = new DiagnosticInfo(
+                                MessageProvider.Instance,
+                                (int)ErrorCode.ERR_TypeForwardedToMultipleAssemblies,
+                                emittedName.FullName,
+                                firstSymbol.Name,
+                                secondSymbol.Name);
+                            return new MissingMetadataTypeSymbol.TopLevelWithCustomErrorInfo(SourceModule, ref emittedName, forwardingErrorInfo);
```

Comment:
Perhaps a static method on `MissingMetadataTypeSymbol` or an instance method on `ModuleSymbol`. Note, it's not strictly a one-liner since the `DiagnosticInfo` can be constructed in the helper too. We should fix this to avoid the code duplication.